sys/net/application_layer/nanocoap/sock.c

```diff
@@ -79,28 +98,31 @@ ssize_t nanocoap_request(coap_pkt_t *pkt, sock_udp_ep_t *local, sock_udp_ep_t *r
                 DEBUG("nanocoap: error parsing packet\n");
                 res = -EBADMSG;
             }
+            else if (coap_get_id(pkt) != id) {
+                res = -EBADMSG;
+                continue;
+            }
+
             break;
         }
     }
 
-    sock_udp_close(&sock);
-
     return res;
 }
 
-ssize_t nanocoap_get(sock_udp_ep_t *remote, const char *path, uint8_t *buf, size_t len)
+ssize_t nanocoap_get(sock_udp_t *sock, const char *path, void *buf, size_t len)
 {
     ssize_t res;
     coap_pkt_t pkt;
     uint8_t *pktpos = buf;
 
-    pkt.hdr = (coap_hdr_t*)buf;
+    pkt.hdr = buf;
     pktpos += coap_build_hdr(pkt.hdr, COAP_TYPE_CON, NULL, 0, COAP_METHOD_GET, 1);
```

Comment:
Any idea why `id` is always 1 here?

Reply:
That is indeed wrong. The message ID is used to uniquely identify a specific message (for the same sender and received endpoint) for a given timespan, after which it can be reused. If you hardcode 1 here, you might end up reusing it before the timespan is over after which it is safe to reuse.

Reply:
Does it make sense to open an issue and *not* also solve this in this PR?